src/createApp.ts

```diff
@@ -837,6 +852,48 @@ const createApp = compose({
 			return instanceRegistries.get(this).identifyStore(store);
 		},
 
+		createWidget<U extends Child, O>(this: App, factory: ComposeFactory<U, O>, options: any = {}): Promise<[ string, U ]> {
+			const app = this;
+			const { defaultWidgetStore, registryProvider } = this;
+			const { id = generateWidgetId() } = options;
```

Comment:
Correct, they are wholly decoupled, intentionally.  It is specifically designed that you don't get the "store magic" when you do not specify and ID.  Widgets, if not supplied at creation, only autogenerate an ID upon render, and are unique to the page.  Auto generated app widget IDs need to be unique to the app.  Since we have two different domains, we have to keep the possibility of collisions.  Mark and I talked a bit about how to not make hard assumptions between the two IDs, even if the normal use case should never trigger this logic.